### PR TITLE
Handling not assigned projects when filtering by DO

### PR DESCRIPTION
**Pull request by dneed-nimble**

### CODE REVIEW

File: TramsDataApi/Gateways/AcademyConversionProjectGateway.cs [hunk 1]
```diff
@@ -76,6 +76,11 @@ public async Task<PagedResult<AcademyConversionProject>> SearchProjects(int page
             {
                 var lowerDeliveryOfficers = deliveryOfficers.Select(officer => officer.ToLower());
                 academyConversionProjects = academyConversionProjects.Where(acp => lowerDeliveryOfficers.Contains(acp.AssignedUserFullName.ToLower()));
+                if (lowerDeliveryOfficers.Contains("Not assigned"))
+                {
+                    var notAssignedProjects = academyConversionProjects.Where(acp => acp.AssignedUserFullName == "" && acp.AssignedUserFullName == null);
```
Comment (jordan-beard) [inline TramsDataApi/Gateways/AcademyConversionProjectGateway.cs:81]: You can shorten this to string.IsNullOrEmpty()

File: TramsDataApi/Gateways/AcademyConversionProjectGateway.cs [hunk 2]
```diff
@@ -76,6 +76,11 @@ public async Task<PagedResult<AcademyConversionProject>> SearchProjects(int page
...
+                    academyConversionProjects = academyConversionProjects.Concat(notAssignedProjects);
```
Comment (jordan-beard) [inline TramsDataApi/Gateways/AcademyConversionProjectGateway.cs:82]: Now that we are combining to sets of results you may need to order at the end rather than the beginning.

File: TramsDataApi/Gateways/AcademyConversionProjectGateway.cs (hunk 1 above)
Comment (dneed-nimble) [reply TramsDataApi/Gateways/AcademyConversionProjectGateway.cs]: Done

File: TramsDataApi/Gateways/AcademyConversionProjectGateway.cs [hunk 3]
```diff
@@ -76,6 +76,11 @@ public async Task<PagedResult<AcademyConversionProject>> SearchProjects(int page
             {
                 var lowerDeliveryOfficers = deliveryOfficers.Select(officer => officer.ToLower());
                 academyConversionProjects = academyConversionProjects.Where(acp => lowerDeliveryOfficers.Contains(acp.AssignedUserFullName.ToLower()));
+                if (lowerDeliveryOfficers.Contains("Not assigned"))
```
[the remark this answers is not in the document]
Comment (dneed-nimble) [reply TramsDataApi/Gateways/AcademyConversionProjectGateway.cs:79]: Done

File: TramsDataApi/Gateways/AcademyConversionProjectGateway.cs (hunk 2 above)
Comment (dneed-nimble) [reply TramsDataApi/Gateways/AcademyConversionProjectGateway.cs]: Done
